test/test_transforms_v2_refactored.py

```diff
@@ -308,6 +308,82 @@ def wrapper(input, *args, **kwargs):
     return wrapper
 
 
+def make_input(input_type, *, dtype=None, device="cpu", spatial_size=(17, 11), mask_type="segmentation", **kwargs):
+    if input_type in {torch.Tensor, PIL.Image.Image, datapoints.Image}:
+        input = make_image(size=spatial_size, dtype=dtype or torch.uint8, device=device, **kwargs)
+        if input_type is torch.Tensor:
+            input = input.as_subclass(torch.Tensor)
+        elif input_type is PIL.Image.Image:
+            input = F.to_image_pil(input)
+    elif input_type is datapoints.BoundingBox:
+        kwargs.setdefault("format", datapoints.BoundingBoxFormat.XYXY)
```

Comment:
Shouldn't this logic be part of `make_bounding_box`'s contract? (same for the `dtype` logic)

This is the kind of changes that makes me think we should already be moving away from the utils in `common_utils` for input generation, as suggested earlier https://github.com/pytorch/vision/pull/7562#discussion_r1236702589.

We can delay it but we'll have to do it eventually anyway, and IMO the earlier we do it, the more control we'll have.

Reply:
Will have a go at it after this PR before I port anything more.

Reply:
See #7717